--- a/linux-tkg-config/prepare	2023-01-12 09:36:41.517667800 -0700
+++ b/linux-tkg-config/prepare	2023-01-12 10:00:03.823945387 -0700
@@ -410,23 +410,6 @@
 
 _tkg_initscript() {
 
-  typeset -Ag _deprecated_config_var_set
-
-  # Check if user is defining deprecated config vars
-  _deprecated_config_var_set=(
-    ["_use_tmpfs"]=`[ -n "$_use_tmpfs" ]; echo $?`
-    ["_source_in_tmpfs"]=`[ -n "$_source_in_tmpfs" ]; echo $?`
-    ["_tmpfs_path"]=`[ -n "$_tmpfs_path" ]; echo $?`
-  )
-
-  for _deprectated_config_var in "${!_deprecated_config_var_set[@]}"; do
-    if [ "${_deprecated_config_var_set[$_deprectated_config_var]}" == "0" ]; then
-      warning "The deprecated config var $_deprectated_config_var has been set"
-      warning "Please check the latest customization.cfg file to see what replaces it"
-      exit 1
-    fi
-  done
-
   # Default to Arch
   if [ -z "$_distro" ] || [ "$_ispkgbuild" = "true" ]; then
     msg2 "Defaulting to Archlinux target\n"
