--- a/linux-tkg-config/prepare.orig	2021-10-06 11:50:17.231084317 -0700
+++ b/linux-tkg-config/prepare	2021-10-06 11:54:47.988008407 -0700
@@ -1411,47 +1411,10 @@
     fi
 
     if [ 0 -lt "${#fragments[@]}" ]; then
-      scripts/kconfig/merge_config.sh -m .config "${fragments[@]}"
+      scripts/kconfig/merge_config.sh .config "${fragments[@]}"
     fi
   fi
 
-  # Distro specific workarounds in the .config file, when using Arch config file by default
-  if [ -z  "$_configfile" ] || [ "$_configfile" = "config_hardened.x86_64" ]; then
-    if [[ "$_distro" =~ ^(Debian|Ubuntu)$ ]]; then
-      _disable "MODULE_COMPRESS_ZSTD"
-      _enable "MODULE_COMPRESS_NONE"
-    fi
-  fi
-
-  # set _menuconfig early for Void
-  if [ "$_distro" = "Void" ]; then
-    _menuconfig="Void"
-  fi
-
-  # rewrite configuration
-  msg2 "Setting config"
-  make ${_config_updating} ${llvm_opt} |& tee -a "$_where"/logs/prepare.log.txt
-
-  # menuconfig / nconfig
-  if [ -z "$_menunconfig" ] && [ "$_distro" != "Void" ]; then
-    plain ""
-    plain "*Optional* For advanced users - Do you want to use make menuconfig or nconfig"
-    plain "to configure the kernel before building it?"
-    plain "If you do, make sure your terminal is currently"
-    plain "at least 19 lines by 80 columns large or you'll get an error :D"
-    read -rp "`echo $'    > 0. nope\n      1. menuconfig\n      2. nconfig\n      3. xconfig\n      choice[0-3?]: '`" CONDITIONMNC;
-    _menunconfig="$CONDITIONMNC"
-  fi
-  if [ 1 = "$_menunconfig" ]; then
-    cp .config .config.orig
-    make menuconfig ${llvm_opt}
-  elif [ 2 = "$_menunconfig" ]; then
-    cp .config .config.orig
-    make nconfig ${llvm_opt}
-  elif [ 3 = "$_menunconfig" ]; then
-    cp .config .config.orig
-    make xconfig ${llvm_opt}
-  fi
   if [ 1 = "$_menunconfig" ] || [ 2 = "$_menunconfig" ] || [ 3 = "$_menunconfig" ]; then
     if [ -z "${_diffconfig}" ]; then
       while true; do
